man/fi_pingpong.1.md

```diff
@@ -0,0 +1,148 @@
+---
+layout: page
+title: fi_pingpong(1)
+tagline: Libfabric Programmer's Manual
+---
+{% include JB/setup %}
+
+
+# NAME
+
+fu_pingpong  \- Quick and simple pingpong test for libfabric
+
+
+# SYNOPSYS
+
+```
+ fi_pingpong [OPTIONS] -s <server IP>	start server
+ fi_pingpong [OPTIONS] <server IP>	connect to server
+```
+
+
+# DESCRIPTION
+
+`fi_pingpong` is a generic pingpong test allowing provider authors and libfabric users to test
+the core feature of the libfabric library: transmitting data between two nodes.
+Moreover it agregates some data during the test on each end in order to diplay some statistics
+after each test has been run, and potentially ask to verify data consistency.
+
+An endpoint can be specified to be used as the data endpoint. By default, the DGRAM endpoint
+is always selected. Therefore it is designed to transmit data from one node toward the other
+over the selected endpoint, or the default DGRAM if none is selected.
+
+
+# HOW TO RUN TESTS
+
+To launch properly a pingpong test, two termimals have to be open : one for the client,
+the other for the server. However the server must be already running to start the client.
+
+As a client-server test, each have the following usage model:
+
+server$	fi_pingpong -s <server endpoint address>	: start server
+client$	fi_pingpong <server endpoint address>		: connect to server
+
+
+# OPTIONS
+
+The command line of the server and the client must be launched with exactly the same set of options. Otherwise the test would fail because the two nodes would not be able to communicate properly.
+
+## Nodes addressing
+
+*-b <src_port>*
+: The non-default source port number of the endpoint.
+
+*-p <dest_port>*
+: The non-default destination port number of the endpoint.
+
+*-s <src_addr>*
+: The source address.
+
+## Fabric
+
+*-f <provider_name>*
+: The name of the underlying fabric provider e.g. sockets, verbs, psm etc. If the provider name is not provided, the test will pick one from the list of the available providers it finds by fi_getinfo call.
+
+*-e <endpoint> where endpoint = (dgram|rdm|msg)*
+The type of endpoint to be used for data messaging between the two nodes.
+
+*-n <domain>*
+: The name of the specific domain to be used.
+
+## Messaging
+
+*-I <iter>*
+: The number of iterations of the test will run.
+
+*-S <msg_size>*
+: The specific size of the message in bytes the test will use or 'all' to run all the default sizes.
+
+## Utils
+
+*-v*
+: Activate the verification of incoming data.
```

Comment:
Activate data integrity checks at the receiver (note: this may have a performance impact).